lib/data.js

```diff
@@ -74,19 +102,7 @@ function duplicateArray(arr, references) {
   const size = arr.length;
   const array = new Array(size);
   references.set(arr, array);
-  let i, val;
-  for (i = 0; i < size; i++) {
-    val = arr[i];
-    if (references.has(val)) {
-      array[i] = references.get(val);
-    } else  if (typeof(val) !== 'object' || val === null) {
-      array[i] = val;
-    } else if (Array.isArray(val)) {
-      array[i] = duplicateArray(val, references);
-    } else {
-      array[i] = duplicateObject(val, references);
-    }
-  }
+  duplicateFields(references, arr, array, range(0, size - 1));
```

Comment:
I don't think generating new array using `range` and then iteration through this array will be faster than `for-loop` through integer indexes of array.